presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1155,6 +1158,13 @@ else if (frame.getType() == GROUPS) {
                 }
             }
 
+            if (node.isIgnoreNulls()) {
```

Comment:
Most of the window function semantic checks happen here https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java#L981. Any particular reason we didn't add this check after the frame validation in that block ?

@tdcmeehan

Reply:
There wasn't any reason before that I had placed it on that exact line, but it needs to be after the function variable has been initialized. I moved it back to line 1122.